stl/inc/numeric

```diff
@@ -461,6 +461,23 @@ _FwdIt2 transform_inclusive_scan(_ExPo&& _Exec, _FwdIt1 _First, _FwdIt1 _Last, _
     _UnaryOp _Transform_op) noexcept; // terminates
 #endif // _HAS_CXX17
 
+template <class _Ty>
+_CONSTEXPR20 bool _Arrays_overlap(const _Ty* const _First, const _Ty* const _Second, const size_t _Common_count) {
+    const uintptr_t _First_val  = reinterpret_cast<uintptr_t>(_First);
+    const uintptr_t _Second_val = reinterpret_cast<uintptr_t>(_Second);

```

Comment:
`reinterpret_cast` is not constexpr-friendly.

Sadly, I'm not aware of any standard manner to determine `_Arrays_overlap` in constant evaluation without stupid loops. We can do this better with `__bulitin_constant_p` for Clang, but there not yet good thing for MSVC.

Reply:
Thanks, I'll add `!_STD is_constant_evaluated()` instead of trying to get that to `_CONSTEXPR20`.

(We're also not restricted to the standard, it would be good to attempt, but otherwise we're good if we invent MSVC-friendly way)

Reply:
Do you need to actually cast to uintptr instead of just dealing with pointers?

Can't you just check _First + count > second? Technically only std::greater is blessed to do it but we know that MSVC implements that for raw pointer ordering too so can just do it directly I feel

Reply:
Apparently you're right. I thought it was Undefined Behavior to compare pointers to potentially different arrays, but it is just unspecified, according to [[expr.rel]/4](https://eel.is/c++draft/expr.compound#expr.rel-4). It is pointer subtraction that is undefined for unrelated arrays.

Reply:
Replaced with usual pointer math.
Still not doing it in constexpr: MSVC behaves strange, other compilers reject because unspecified, see https://godbolt.org/z/j7oehcz6s